R/helpers_jags.R

```diff
@@ -56,23 +56,25 @@ h_jags_add_dummy <- function(object, where, dummy = 0) {
 #'   function bodies are not preserved in any way after joining.
 #'
 #' @param model1 (`function`)\cr the first model to join.
-#' @param model2 (`function`)\cr the second model to join.
+#' @param model2 (`function` or `call`)\cr the second model to join.
 #'
 #' @return joined models.
 #'
 #' @export
 #'
 h_jags_join_models <- function(model1, model2) {
+  if (is.function(model2)) model2 <- body(model2)
+
   assert_function(model1)
-  assert_function(model2)
-  assert_class(body(model1), "{")
-  assert_class(body(model2), "{")
-
-  body2 <- as.list(body(model2))
-  if (length(body2) >= 2) {
-    body1 <- as.list(body(model1))
-    body(model1) <- as.call(c(body1, body2[-1]))
+  assert_true(is.language(model2))
+
+  if (inherits(model2, "{")) {
+    model2_exprs <- as.list(model2)[-1]
+  } else {
+    model2_epxrs <- as.list(model2)
   }
+
+  body(model1) <- as.call(append(as.list(body(model1)), model2_exprs))
   model1
 }
```

Comment:
This is the primary change of the patch, accepting `model2` as a language object instead of strictly as a function.

To try to keep it backwards compatible through the refactor, it will still allow functions. I would probably encourage abandoning the function interface in its entirety in favor of passing expressions, but I'm not familiar enough with the use patterns to really know whether that would be worth the refactor.

Reply:
Thanks Doug, it would indeed great to have backwards compatibility re: extensions of the R package by the user, so to allow them the use of functions still